src/modules/sensors/vehicle_air_data/VehicleAirData.cpp

```diff
@@ -110,6 +110,22 @@ void VehicleAirData::SensorCorrectionsUpdate(bool force)
 	}
 }
 
+void VehicleAirData::AirTemperatureUpdate()
+{
+	differential_pressure_s differential_pressure;
+
+	static constexpr float temperature_min_celsius = -20.f;
+	static constexpr float temperature_max_celsius = 35.f;
+
+	// update air temperature if data from differential pressure sensor is finite and not exactly 0
+	// limit the range to max 35°C to limt the error due to heated up airspeed sensors prior flight
+	if (_differential_pressure_sub.update(&differential_pressure) && PX4_ISFINITE(differential_pressure.temperature)
+	    && fabsf(differential_pressure.temperature) > FLT_EPSILON) {
```

Comment:
Where is 0 coming from? Ideally the drivers should now be setting temperature to NAN if they don't have a reading.

Reply:
We have one that doesn't. But if that's a one off than we should fix it in there, agree. So you would assess the chance of a driver publishing 0 instead of NAN to be small?